Sprint_6

smaikles opened the pull request.

Commit:
Мой первый комит.

Создал ветку develop. Наполнил файл pom, подключил зависимости. Обновил файл gitignore

Commit:
Второй комит.

Написал тесты для класса Cat.
плагин Jacoco  показывает что класс Cat покрыт на 100%

Commit:
Третий комит.

Поправил файл pom.xml (была ошибка из-за которой не читалась кириллица)
Написал тесты для класса Feline.
Сделал правку в классе Lion (убрал зависимости)
Подготовил тестовый класс для Lion

Commit:
add jacoco report

Commit:
Пятый комит.

Поправил класс Lion (в классе была грамматическая ошибка)
Написал тесты для Lion
плагин jacoco  показывает покрытие всего кода на 92%

digranate commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,79 @@
+            Assert.assertEquals(hasMane, lion.doesHaveMane());
+        } catch (Exception e) {
+            errorMessageCheck(e);
+        }
```
Так не получится. Если у тебя на Другое метод выдаст false, то тест не упадёт, а должен упасть. И обратно если на Самке выпадет исключение, то тоже тест это проглотит.
Нужно реализовать по-другому. Тут оставить только позитивные кейсы. Негативные вынести в другой класс и сделать через аннотацию или с помощью assertThrows.

digranate commented on src/test/java/LionTest.java:
```diff
@@ -0,0 +1,79 @@
+        try {
+            Lion lion = new Lion(sex, feline);
+            List<String> expectedList = List.of("Животные", "Птицы", "Рыба");
+            Mockito.when(lion.getFood()).thenReturn(expectedList);
```
Мы никогда не подменяем результат того метода, который сейчас тестируем. Подменяем только зависимости.

pomegranate282922 reviewed (commented):
Изоляция выполнена верно!  В тестах есть 2 проблемки 1. Подменяешь результат того метода, который сейчас тестируешь 2. Тесты на исключения нужно переделать

Commit:
Исправленный комит.

Динара, доброе утро :) Понял принял, постарался сейчас убрать ошибки.
Изменил тесты для Lion, исправил замечания
плагин jacoco  показывает покрытие всего кода на 92%

pomegranate282922 reviewed (approved):
Все хорошо!
А тесты на исключение решил вообще не делать?) Зачёт, так как это опционально.
